From 89f34f16c5c1537248d1d953ffafc70507798e9a Mon Sep 17 00:00:00 2001
From: Yaroslav Halchenko <debian@onerussian.com>
Date: Wed, 1 Nov 2017 19:11:33 -0400
Subject: [PATCH 2/4] BF: plot_dataset_chunks - do not invoke ion in non TkAgg
 backends

lead to it getting stuck... now also close the current figure right after plotting
speeds things up etc
---
 mvpa2/misc/plot/base.py       | 5 +++--
 mvpa2/tests/test_misc_plot.py | 2 ++
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/mvpa2/misc/plot/base.py b/mvpa2/misc/plot/base.py
index cd87de09..78f198cd 100644
--- a/mvpa2/misc/plot/base.py
+++ b/mvpa2/misc/plot/base.py
@@ -582,7 +582,7 @@ def plot_dataset_chunks(ds, clf_labels=None):
     incorrectly labeled samples will have 'x' in them
     """
     if ds.nfeatures != 2:
-        raise ValueError, "Can plot only in 2D, ie for datasets with 2 features"
+        raise ValueError("Can plot only in 2D, ie for datasets with 2 features")
     if pl.matplotlib.get_backend() == 'TkAgg':
         pl.ioff()
     if clf_labels is not None and len(clf_labels) != ds.nsamples:
@@ -615,7 +615,8 @@ def plot_dataset_chunks(ds, clf_labels=None):
             1.1 * np.max(dss[:, 0]),
             1.1 * np.min(dss[:, 1])))
     pl.draw()
-    pl.ion()
+    if pl.matplotlib.get_backend() == 'TkAgg':
+        pl.ion()
 
 
 def timeseries_boxplot(median, mean=None, std=None, n=None, min=None, max=None,
diff --git a/mvpa2/tests/test_misc_plot.py b/mvpa2/tests/test_misc_plot.py
index a61d3e5c..56a47db3 100644
--- a/mvpa2/tests/test_misc_plot.py
+++ b/mvpa2/tests/test_misc_plot.py
@@ -37,6 +37,8 @@ def test_plot_dataset_chunks(dsp):
     # smoke test for now
     if 'chunks' not in ds.sa:
         return  # nothing to plot in this one
+    print dsname
     plot_dataset_chunks(ds[:, :2])  # could only plot two
+    pl.close(pl.gcf())
     if ds.nfeatures > 2:
         assert_raises(ValueError, plot_dataset_chunks, ds)
\ No newline at end of file
-- 
2.14.1

